src/strategies/DebtStrategy.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {AccessStrategy} from "src/strategies/AccessStrategy.sol";
+import {AffineVault} from "src/vaults/AffineVault.sol";
+
+contract DebtStrategy is AccessStrategy {
+    using SafeTransferLib for ERC20;
+
+    // amount of debt strategy has
+    uint256 public debt;
+
+    constructor(AffineVault _vault, address[] memory strategists) AccessStrategy(_vault, strategists) {}
+
+    /**
+     * @notice collect the funds from the investments
+     * @param amount amount needs to be withdraw
+     * @return amount withdrawn
+     */
+    function divest(uint256 amount) external override onlyVault returns (uint256) {
+        // withdraw funds
+        uint256 divestedAmount = _divest(amount);
+
+        // settle debt
+        // debt+amount is total debt as there might be existing debt.
+        if (asset.balanceOf(address(this)) < (debt + amount)) {
```

Comment:
The debt can only go up, not down, during a divestment.

Increase the `debt` by `amount - divestedAmount` if `amount > divestedAmount`